dune/iga/nurbstrimfunctionality.hh

```diff
@@ -0,0 +1,303 @@
+//
+// Created by Henri on 14.03.2023.
+//
+
+#pragma once
+
+namespace Dune::IGA::Impl::Trim {
+
+  using ctype        = double;
+  using ClipperPoint = Clipper2Lib::Point<ctype>;
+  using Point        = FieldVector<ctype, 2>;
+  using PointVector  = std::vector<Point>;
+
+  // To keep track of which point belongs to which edge or node we will use a pointMap
+  struct IntersectionPoint {
+    Point point;
+    bool alreadyVisited = false;
+
+    explicit IntersectionPoint(ClipperPoint& _clipperPoint) : point({_clipperPoint.x, _clipperPoint.y}){};
+  };
+
+  using IntersectionPointMap    = std::map<int, std::vector<IntersectionPoint>>;
+  using IntersectionPointMapPtr = std::shared_ptr<std::map<int, std::vector<IntersectionPoint>>>;
+
+  PointVector getElementCorners(const auto& element) {
+    auto geo = element.geometry();
+
+    std::vector<Point> corners(4);
+    corners[0] = geo.corner(0);
+    corners[1] = geo.corner(1);
+    corners[2] = geo.corner(3);  // see dune book page 127 Figure 5.12
+    corners[3] = geo.corner(2);
+
+    return corners;
+  }
+
+  Clipper2Lib::PathD getElementEdges(const auto& element) {
+    auto corners = getElementCorners(element);
+    Clipper2Lib::PathD edges;
+    for (int i = 0; i < 4; ++i)
+      edges.emplace_back(corners[i][0], corners[i][1]);
+    return edges;
+  }
+
+  bool areCurvesOnTheSameLine(NurbsCurveHandler& curve1, NurbsCurveHandler& curve2) {
```

Comment:
Strange naming maybe, for me it is unclear what does it mean if curves "are" on the same line? Do they have the same intersection point with the line, do they both have some intersection point ? or are they just the same curves.

From the function body i think you want to check of the two lines that conncet the end points of the two curves are colinear.

Wouldn't it be sufficient to just check `Dune::Floatcmp:eq(cos_angle,1.0)` ?